File: Code 3 Callouts/Code 3 Callouts/Models/Interiors/Interior.vb
Imports LSPD_First_Response.Engine
Imports Rage
Imports Rage.Native

Namespace Models.Interiors

    Public Class Interior
        Public Sub New()
            Init()
        End Sub

        Public Sub New(ByVal pType As InteriorType, ByVal pInteriorSpawnPoint As SpawnPoint, ByVal pHidingPlaces As List(Of SpawnPoint))
            Type = pType
            InteriorSpawnPoint = pInteriorSpawnPoint
            HidingPlaces = pHidingPlaces
        End Sub

        Private Sub Init()
            Type = InteriorType.Null
            InteriorSpawnPoint = SpawnPoint.Zero
            HidingPlaces = New List(Of SpawnPoint)
        End Sub

        Public Sub LoadInterior()
            If InteriorSpawnPoint <> SpawnPoint.Zero Then
                Dim mInterior As Integer = NativeFunction.CallByName(Of Integer)("GET_INTERIOR_AT_COORDS", InteriorSpawnPoint.Position.X, InteriorSpawnPoint.Position.Y, InteriorSpawnPoint.Position.Z)
                NativeFunction.CallByHash(Of UInteger)(&H2CA429C029CCF247UL, mInterior)
                NativeFunction.CallByName(Of UInteger)("SET_INTERIOR_ACTIVE", mInterior, True)
                NativeFunction.CallByName(Of UInteger)("DISABLE_INTERIOR", mInterior, False)
            End If
        End Sub

        Public Property Type As InteriorType
        Public Property InteriorSpawnPoint As SpawnPoint
        Public Property HidingPlaces As List(Of SpawnPoint)
    End Class

End Namespace

File: Code 3 Callouts/Code 3 Callouts/Models/Interiors/Interior.cs
using LSPD_First_Response.Engine;
using Rage;
using Rage.Native;
using System.Collections.Generic;
using static Stealth.Plugins.Code3Callouts.Models.Interiors.InteriorDatabase;

namespace Stealth.Plugins.Code3Callouts.Models.Interiors
{

	internal class Interior
	{
		internal Interior()
		{
			Init();
		}

		internal Interior(InteriorType pType, SpawnPoint pInteriorSpawnPoint, List<SpawnPoint> pHidingPlaces)
		{
			Type = pType;
			InteriorSpawnPoint = pInteriorSpawnPoint;
			HidingPlaces = pHidingPlaces;
		}

		private void Init()
		{
			Type = InteriorType.Null;
			InteriorSpawnPoint = SpawnPoint.Zero;
			HidingPlaces = new List<SpawnPoint>();
		}

		internal void LoadInterior()
		{
			if (InteriorSpawnPoint != SpawnPoint.Zero) {
				int mInterior = NativeFunction.Natives.GetInteriorAtCoords<int>(InteriorSpawnPoint.Position.X, InteriorSpawnPoint.Position.Y, InteriorSpawnPoint.Position.Z);
                NativeFunction.Natives.x2ca429c029ccf247(mInterior); // _LOAD_INTERIOR
                NativeFunction.Natives.SetInteriorActive(mInterior, true);
                NativeFunction.Natives.DisableInterior(mInterior, false);
			}
		}

		internal InteriorType Type { get; set; }
		internal SpawnPoint InteriorSpawnPoint { get; set; }
		internal List<SpawnPoint> HidingPlaces { get; set; }
	}

}